packages/generator/src/v1.1/generateBulkActions.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2023 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import path from "node:path";
+import type { MinimalFs } from "../MinimalFs";
+import { getModifiedEntityTypes } from "../shared/getEditedEntities";
+import { formatTs } from "../util/test/formatTs";
+import type { WireOntologyDefinition } from "../WireOntologyDefinition";
+import { getTypeScriptTypeFromDataType } from "./generateActions";
+
+export async function generateBulkActions(
+  ontology: WireOntologyDefinition,
+  fs: MinimalFs,
+  outDir: string,
+  importExt: string = "",
+) {
+  const importedObjects = new Set<string>();
+  let actionSignatures: any[] = [];
+  for (const action of Object.values(ontology.actionTypes)) {
+    const entries = Object.entries(action.parameters);
+
+    const modifiedEntityTypes = getModifiedEntityTypes(action);
+    const addedObjects = Array.from(modifiedEntityTypes.addedObjects);
```

Comment:
I think `.addedObjects`/`.modifiedObjects` is a `Set<string>`. So we are converting it to an array so we can `forEach` add it to a different set. At the very least you could just `forEach` on the returned sets instead of the memory copy.

Reply:
Ah yea I just pulled this from the `generateActions`, it looks like we converted to Array so we could just do a simple join on line 74 when generating the edits return type